README.md

```diff
@@ -400,6 +401,68 @@ Example of creating regions in interleaved mode, using all NVDIMMs:
 # ipmctl create -goal PersistentMemoryType=AppDirect
 ```
 
+When running inside virtual machines, each virtual machine typically
+already gets access to one region and `ipmctl` is not needed inside
+the virtual machine. Instead, that region must be made available for
+use with `ndctl` because when the virtual machine comes up for the
+first time, the entire region is already allocated for use as a single
+block device:
+``` sh
+$ $ ndctl list -RN
+{
+  "regions":[
+    {
+      "dev":"region0",
+      "size":34357641216,
+      "available_size":0,
+      "max_available_extent":0,
+      "type":"pmem",
+      "persistence_domain":"unknown",
+      "namespaces":[
+        {
+          "dev":"namespace0.0",
+          "mode":"raw",
+          "size":34357641216,
+          "sector_size":512,
+          "blockdev":"pmem0"
+        }
+      ]
+    }
+  ]
+}
+$ ls -l /dev/pmem*
+brw-rw---- 1 root disk 259, 0 Jun  4 16:41 /dev/pmem0
+```
+
+Labels must be initialized in such a region, which must be performed
+once after the first boot:
+``` sh
+$ sudo ndctl disable-region region0
+disabled 1 region
+$ sudo ndctl init-labels nmem0
+initialized 1 nmem
+$ sudo ndctl enable-region region0
+enabled 1 region
+$ ndctl list -RN
+[
+  {
+    "dev":"region0",
+    "size":34357641216,
+    "available_size":34357641216,
+    "max_available_extent":34357641216,
+    "type":"pmem",
+    "iset_id":10248187106440278,
+    "persistence_domain":"unknown"
+  }
+]
+$ ls -l /dev/pmem*
+ls: cannot access '/dev/pmem*': No such file or directory
+```
+
+Beware that the PMEM-CSI driver will run without errors on a node
+where PMEM that was not prepared as described above. It will then
+report zero local storage for that node, something that currently is
+only visible in the log files.
```

Comment:
thats good point to raise about current driver startup behavior in LVM mode -- should driver refuse to run if there is no available_space in region and no namespaces for driver use exist? because running driver in such state will never produce anything useful on that node, should driver instead "fail early" and show the problem in exit status. That would be more operator-friendly than keeping running, nerver serving anything. and letting operator to find out the reason through log files analysis and examining node state (using ndctl). It can be estimated situations like those will happen in the field.

Reply:
actually, that specific path to trouble on a clean system is qemu specific. To reach such situation on a HW-based node (i.e. what I call "field scenario") there must be previously used regions/namespaces involved, i.e. not so common path.

Reply:
What is the situation on real hardware with a "fresh" PMEM NVDIMM when `ipmctl create` wasn't called? Is that resulting in a failed driver pod or will it run without memory?

Either way, yes, we need to consider how to surface these setup problems through the Kubernetes API. My plan was to make it part of the PMEM CSI operator, because that then can have a status field and events that show when something is amiss.

For now let's keep everything as it is.

Reply:
out-of-box NVDIMM will be in Memory mode before ipmctl goal creation (which needs reboot to activate). With NVDIMM device(s) in memory mode, no PMEM storage is available for PMEM-CSI driver, it will start and run, but will fail to create volumes. That's similar to running on any host: driver will run but can not serve volumes. That's because there is no HW detection in driver: it can't know are there device(s) present. The driver can see Regions and Namespaces over libndctl API, and these appear after "ipmctl create goal + reboot".

Reply:
So it *is* exactly as in the VM case, which is what I thought.

Now I am confused by your earlier statement that "that specific path to trouble on a clean system is qemu specific". Hmm, perhaps it wasn't clear that with "PMEM not prepared as described above" I meant both cases?

This confusion could be avoided by moving that warning into the first paragraph, directly after "The PMEM-CSI driver itself cannot create such regions."

Reply:
Although the end result in driver is same, states (and path to state) are different and should not be thought as same thing:
- HW: no regions exist (ipmctl was not used), driver will not find usable space.
- qemu: kernel pmem subsystem prepares one region and fills it with namespace -> driver will not find usable space.

The paths forward, to solve it, are different as well:
- HW: use ipmctl create goal, reboot -> driver will find regions with all space free, will create namespaces
- qemu: use 3 labeling commands to redo region formatting, then driver will see one region with space

That is what I meant with "path to trouble is qemu specific". On HW, you will end up with space-taken-by-existing-namespace(s) only via previous use. On qemu, this is the state after VM creation.

Reply:
Ack. I pushed another commit where I move the error message. I think it is clearer this way and less likely to be missed.